website/common/script/libs/getItemInfo.js

```diff
@@ -9,12 +9,27 @@ import isFreeRebirth from './isFreeRebirth';
 import getOfficialPinnedItems from './getOfficialPinnedItems';
 
 function lockQuest (quest, user) {
-  if (quest.key === 'lostMasterclasser1') return !(user.achievements.quests.dilatoryDistress3 && user.achievements.quests.mayhemMistiflying3 && user.achievements.quests.stoikalmCalamity3 && user.achievements.quests.taskwoodsTerror3);
+  // checks series quests, including Masterclasser
+  if (quest.prereqQuests) {
+    if (!user.achievements.quests) return true;
+    for (const prereq of quest.prereqQuests) {
+      if (!user.achievements.quests[prereq]) return true;
+    }
+  }
+  // checks quest & user level against quest level
   if (quest.lvl && user.stats.lvl < quest.lvl) return true;
-  if (quest.unlockCondition && (quest.key === 'moon1' || quest.key === 'moon2' || quest.key === 'moon3')) {
-    return user.loginIncentives < quest.unlockCondition.incentiveThreshold;
+
+  // checks unlockCondition.incentiveThreshold
+  if (
+    quest.unlockCondition
+    && quest.unlockCondition.incentiveThreshold
+    && user.loginIncentives < quest.unlockCondition.incentiveThreshold
+  ) return true;
+  if (user.achievements.quests) {
+    return quest.previous
```

Comment:
After this update, do we even need to reference `quest.previous`? Given that all those should be in `quest.prereqQuests` now?

Reply:
Probably not, we could likely change it to `return true`--I'll test that out.